### PR TITLE
Swiftype-powered autocomplete and search

### PR DESCRIPTION
**Pull request by hamedahmadi**
This pull request adds autocomplete and search powered by Swiftype.

To get it set up locally:

1. Create an account at Swiftype.com
2. Create an API-based search engine with a document type called "rubygem"
3. Add your API key (located on the Account Settings page) and Engine key (located on the Engine overview page) as environment variables named like so:

```
ENV['SWIFTYPE_API_KEY'] = "your_api_key"
ENV['SWIFTYPE_ENGINE_KEY'] = "your_engine_key"
ENV['SWIFTYPE_ENGINE_SLUG'] = "your_engine_slug"
```

Presumably this is done via the secret.rb file, but this code is (understandably) not in the repository.

4. Index the gems in your local database: `rake swiftype:index_rubygems`
5. Browse to the home page and try a search.

Once you are happy with this, to go to production, repeat the steps of creating an engine (we recommend having separate engines for each environment/developer) and adding the `SWIFTYPE_API_KEY` and `SWIFTYPE_ENGINE_KEY` to the environment. Upon deploying, you can run the rake task to index all the existing gems and the search will be live.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,4 @@ tmp/
 tags
 .idea
 chef
+.DS_Store
```
Comment (simi) [inline .gitignore]: [global gitgnore](http://robots.thoughtbot.com/global-gitignore)

**Comment by davefp** [comment]
@hamedahmadi As a result of #610 a large chunk of the repo's history was re-written. Please rebase your PR so that it will merge cleanly. Thanks!

(There are further details on the blog: http://blog.rubygems.org/2015/02/01/rewriting-history.html)

**Comment by hamedahmadi** [comment]
@davefp @dwradcliffe I updated this long standing PR. I believe this adds really good autocomplete and search experience to the site. Please LMK if you have any questions.

**Comment by arthurnn** [comment]
@dwradcliffe should we close this in favor of https://github.com/rubygems/rubygems.org/pull/1085

**Pull request closed without merging**